pkg/common/cns-lib/volume/manager.go

```diff
@@ -144,6 +146,17 @@ func (m *defaultManager) ResetManager(ctx context.Context, vcenter *cnsvsphere.V
 	log.Infof("Done resetting volume.defaultManager")
 }
 
+// GetVCVersion helps return vCenter version manager is interacting with
+func (m *defaultManager) GetVCVersion(ctx context.Context) (string, error) {
+	log := logger.GetLogger(ctx)
+	log.Debug("Checking the vCenter version")
+	if m.virtualCenter == nil || m.virtualCenter.Client == nil {
```

Comment:
Can you print `m.virtualCenter` in the logs ?

Reply:
> What if, only m.virtualCenter.Client is nil.

In that case, also we will be returning an error. we do initialize the client at the beginning of the driver and metadata syncer by calling `func GetManager(ctx context.Context, vc *cnsvsphere.VirtualCenter) Manager`.

```
	if m.virtualCenter == nil || m.virtualCenter.Client == nil {
		return "", errors.New("cannot retrieve vcenter version. virtualCenter is not initialized")
	}
```

I can change the log message

```
	if m.virtualCenter == nil || m.virtualCenter.Client == nil {
		return "", errors.New("cannot retrieve vcenter version. virtualCenter or virtualCenter.client is not initialized")
	}
```



>  Can you print m.virtualCenter in the logs ? It could be useful to triage during erroneous initialization

I can not print `m.virtualCenter` in the log, as it contains username and password and other secure details.